pipework

```diff
@@ -205,43 +205,43 @@ else
 fi
 
 # Check if an incompatible VLAN device already exists
-[ $IFTYPE = phys ] && [ "$VLAN" ] && [ -d /sys/class/net/$IFNAME.VLAN ] && {
-  [ -z "$(ip -d link show $IFNAME.$VLAN | grep "vlan.*id $VLAN")" ] && {
+[ "$IFTYPE" = phys ] && [ "$VLAN" ] && [ -d "/sys/class/net/$IFNAME.VLAN" ] && {
+  ip -d link show "$IFNAME.$VLAN" | grep -q "vlan.*id $VLAN" || {
     echo "$IFNAME.VLAN already exists but is not a VLAN device for tag $VLAN"
     exit 1
   }
 }
 
 [ ! -d /var/run/netns ] && mkdir -p /var/run/netns
-[ -f /var/run/netns/$NSPID ] && rm -f /var/run/netns/$NSPID
-ln -s /proc/$NSPID/ns/net /var/run/netns/$NSPID
+rm -f "/var/run/netns/$NSPID"
+ln -s "/proc/$NSPID/ns/net" "/var/run/netns/$NSPID"
 
 # Check if we need to create a bridge.
-[ $IFTYPE = bridge ] && [ ! -d /sys/class/net/$IFNAME ] && {
-  [ $BRTYPE = linux ] && {
-    (ip link add dev $IFNAME type bridge > /dev/null 2>&1) || (brctl addbr $IFNAME)
-    ip link set $IFNAME up
+[ "$IFTYPE" = bridge ] && [ ! -d "/sys/class/net/$IFNAME" ] && {
+  [ "$BRTYPE" = linux ] && {
+    (ip link add dev "$IFNAME" type bridge > /dev/null 2>&1) || (brctl addbr "$IFNAME")
+    ip link set "$IFNAME" up
   }
-  [ $BRTYPE = openvswitch ] && {
-    ovs-vsctl add-br $IFNAME
+  [ "$BRTYPE" = openvswitch ] && {
+    ovs-vsctl add-br "$IFNAME"
   }
 }
 
-MTU=$(ip link show $IFNAME | awk '{print $5}')
+MTU=$(ip link show "$IFNAME" | awk '{print $5}')
 # If it's a bridge, we need to create a veth pair
-[ $IFTYPE = bridge ] && {
+[ "$IFTYPE" = bridge ] && {
   LOCAL_IFNAME="v${CONTAINER_IFNAME}pl${NSPID}"
   GUEST_IFNAME="v${CONTAINER_IFNAME}pg${NSPID}"
-  ip link add name $LOCAL_IFNAME mtu $MTU type veth peer name $GUEST_IFNAME mtu $MTU
+  ip link add name "$LOCAL_IFNAME" mtu "$MTU" type veth peer name "$GUEST_IFNAME" mtu "$MTU"
   case "$BRTYPE" in
     linux)
-      (ip link set $LOCAL_IFNAME master $IFNAME > /dev/null 2>&1) || (brctl addif $IFNAME $LOCAL_IFNAME)
+      (ip link set "$LOCAL_IFNAME" master "$IFNAME" > /dev/null 2>&1) || (brctl addif "$IFNAME" "$LOCAL_IFNAME")
       ;;
     openvswitch)
-      ovs-vsctl add-port $IFNAME $LOCAL_IFNAME ${VLAN:+"tag=$VLAN"}
+      ovs-vsctl add-port "$IFNAME" "$LOCAL_IFNAME" "${VLAN:+tag=$VLAN}"
```

Comment:
Shellcheck didn't notice that I missed quoting the alt expansion at the end of the line. I guess since it's an alt expansion, as long as you quote one of the parts, it doesn't matter.